src/Event/EventFactory.cpp

```diff
@@ -1,4 +1,220 @@
 #include "EventFactory.h"
+#include "../VehicleStatus/VehicleStatus.h"
+#include "../Time/Time.h"
+#include "../Location/Location.h"
+#include "Event.h"
+#include <type_traits>
+#include <atomic>
+#include <iostream>
+#include <cmath>
+#include <thread>
 
 namespace LockFreeDispatch {
+
+
+    std::vector<Vehicle *> EventFactory::vehiclesMeetingRequirements(
+            std::vector<Vehicle *> vehicleReqts,
+            std::vector<Vehicle *> orderedVehicles) {
+
+        std::vector<Vehicle*> meetingRequirements;
+
+        for (const auto & vehicle : orderedVehicles)
```

Comment:
```suggestion
        for (const auto vehicle : orderedVehicles)
```
`vehicle` is a pointer, so it is cheap to copy. No need for a reference.